cpp/include/cudf/column/column_view.hpp

```diff
@@ -294,6 +294,26 @@ class column_view : public detail::column_view_base {
    *---------------------------------------------------------------------------**/
   size_type num_children() const noexcept { return _children.size(); }
 
+  /**---------------------------------------------------------------------------*
+   * @brief Constrcuts a slice of column_view as per requested offset and size.
+   * The new column_view will start at offset and will have the size requested.
+   * @throws `cudf::logic_error` if `slice_offset` with `slice_size` goes beyond
+   * the size of the `column_view`.
+   * @throws `cudf::logic_error` if `slice_size` < 0.
+   * @throws `cudf::logic_error` if `slice_offset` < 0.
+   *
+   * As the views can have offsets, so actual start would be offset of `input` +
+   * offset requested.
+   *
+   * @param slice_offset The offset from which the new column_view should start.
+   * @param slice_size The size of the column_view requested from the offset.
+   *
+   * @return unique_ptr<column_view> The unique pointer to the column view
+   * constrcuted as per the offset and size requested.
+   *---------------------------------------------------------------------------**/
+  std::unique_ptr<column_view> slice (size_type slice_offset,
```

Comment:
Since a `column_view` is non-owning, it can just be returned as a stack object. No need to use a `unique_ptr`.
```suggestion
column_view slice (size_type slice_offset,
```

Reply:
Does this need to be a member function? Can't it just be a free function?

Reply:
It would be easier to access children and it will ensure no changes will be made to calling object.

Apart from that not much of difference.

Reply:
I think I'd like to make this a free function if there's not a strong reason against doing so.